ammo-fe/src/network/MockedSocket.ts

```diff
@@ -1,29 +1,29 @@
 // Arguments can be either a callback function (in the case of the toggleRecord for example), or a list of unknowned arguments
-type Arguments = unknown[] | ((recording: boolean) => void);
+type Arguments = unknown[];
 
 export default class MockedSocket {
     private cache: Record<string, (...args: unknown[]) => void> = {};
 
-    private isRecording = true;
+    public isSubbedToBullets = true;
 
     public connected = true;
 
     on = (event: string, listener: (...args: unknown[]) => void): void => {
         this.cache[event] = listener;
     };
 
-    emit = (event: string, args: Arguments): void => {
-        // When we toggle, we save the new value and invoke the callback given just like the server would do
-        if (event === 'toggleRecord') {
-            this.isRecording = !this.isRecording;
-            if (typeof args === 'function') {
-                args(this.isRecording);
-            }
-            return undefined;
+    emit = (event: string, args: Arguments, cbk: unknown): void => {
+        // Sub to bullets channel
+        if (event === 'bullets::sub') {
+            const { sub } = (args as unknown) as { sub: boolean };
+            this.__handleBulletsSub(sub, cbk as (subbed: boolean) => void);
```

Comment:
All this casting seems weird. We pass a generic parameter, and always cast it to something specific